lib/board.rb

```diff
@@ -0,0 +1,57 @@
+class Board
+
+    attr_accessor :cells
+
+    def initialize
+      @cells = Array.new(9, " ")
+    end
+
+    def reset!
+      @cells.clear
+      @cells = Array.new(9, " ")
+    end
+
+    def first_display
+      puts " #{1} | #{2} | #{3} "
+      puts "-----------"
+      puts " #{4} | #{5} | #{6} "
+      puts "-----------"
+      puts " #{7} | #{8} | #{9} "
+    end
+
+    def display
+      puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+      puts "-----------"
+      puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+      puts "-----------"
+      puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+    end
+
+    def position(board_choice)
+      choice = @cells[board_choice.to_i - 1]
+      choice
+    end
+
+    def full?
+      @cells.all? {|location| location == "X" || location == "O"}
+    end
+
+    def turn_count
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```